web_src/js/features/repo-diff.js

```diff
@@ -28,22 +27,25 @@ function initRepoDiffReviewButton() {
       counter.setAttribute('data-pending-comment-number', num);
       counter.textContent = num;
       // Force the browser to reflow the DOM. This is to ensure that the browser replay the animation
-      $reviewBox.removeClass('pulse');
-      $reviewBox.width();
-      $reviewBox.addClass('pulse');
+      reviewBox.classList.remove('pulse');
+      const _ = reviewBox.getBoundingClientRect().width;
+      reviewBox.classList.add('pulse');
```

Comment:
what the heck 😆 does code rely on the side-effects of `getBoundingClientRect`? Surely there must be better ways.

Reply:
[`523552f` (#30176)](https://github.com/go-gitea/gitea/pull/30176/commits/523552fbe2ebe28cc0ebe4da5d5de3eca3595c0b)